File: ios/ExportFiles.m
//
//  ExportFiles.m
//  Core2
//
//  Created by ueno on 2023/04/15.
//

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(ExportFiles, NSObject)
  RCT_EXTERN_METHOD(
    exportTextFile: (NSString)filename
    text: (NSString)text
  )
  RCT_EXTERN_METHOD(
    readTextFile: (RCTPromiseResolveBlock)resolve
    rejecter: (RCTPromiseRejectBlock)reject
  )
  RCT_EXTERN_METHOD(
    exportBackups: (NSString)chanBackup
  )
  RCT_EXTERN_METHOD(
    readSubmarineBackup: (RCTPromiseResolveBlock)resolve
    rejecter: (RCTPromiseRejectBlock)reject
  )
  RCT_EXTERN_METHOD(
    createSubmarineDbFile: (NSString)dbBase64
    resolver: (RCTPromiseResolveBlock)resolve
    rejecter: (RCTPromiseRejectBlock)reject
  )
@end

File: ios/ExportFiles.swift
//
//  ExportFiles.swift
//  Core2
//
//  Created by ueno on 2023/04/15.
//

import Foundation

@objc(ExportFiles)
class ExportFiles : NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc
  func exportTextFile(
    _ filename: String,
    text: String
  ) {
    // ToDo 実装
    print("ExportFiles.exportTextFile not implemented");
  }

  @objc
  func readTextFile(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    // ToDo 実装
    print("ExportFiles. not implemented");
    reject("ExportFiles", "readTextFile not implemented", nil)
  }

  @objc
  func exportBackups(
    _ chanBackup: String
  ) {
    // ToDo 実装
    print("ExportFiles.exportBackups not implemented");
  }

  @objc
  func readSubmarineBackup(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    // ToDo 実装
    print("ExportFiles.readSubmarineBackup not implemented");
    reject("ExportFiles", "readSubmarineBackup not implemented", nil)
  }

  @objc
  func createSubmarineDbFile(
    _ dbBase64: String,
    resolver resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    // ToDo 実装
    print("ExportFiles.createSubmarineDbFile not implemented");
    reject("ExportFiles", "createSubmarineDbFile not implemented", nil)
  }
}
